src/Resources/config/services.php

```diff
@@ -0,0 +1,297 @@
+            ->args([
+                service(AccessTokenManagerInterface::class),
+                service(ClientManagerInterface::class),
+                service(ScopeConverter::class),
+            ])
+        ->alias(AccessTokenRepositoryInterface::class, AccessTokenRepository::class)
+
+        ->set(RefreshTokenRepository::class)
+            ->args([
+                service(RefreshTokenManagerInterface::class),
+                service(AccessTokenManagerInterface::class),
+            ])
+        ->alias(RefreshTokenRepositoryInterface::class, RefreshTokenRepository::class)
+
+        ->set(ScopeRepository::class)
+            ->args([
+                service(ScopeManagerInterface::class),
+                service(ClientManagerInterface::class),
+                service(ScopeConverter::class),
+                service(EventDispatcherInterface::class),
+            ])
+        ->alias(ScopeRepositoryInterface::class, ScopeRepository::class)
+
+        ->set(UserRepository::class)
+            ->args([
+                service(ClientManagerInterface::class),
+                service(EventDispatcherInterface::class),
+                service(UserConverter::class),
+            ])
+        ->alias(UserRepositoryInterface::class, UserRepository::class)
+
+        ->set(AuthCodeRepository::class)
+            ->args([
+                service(AuthorizationCodeManagerInterface::class),
+                service(ClientManagerInterface::class),
+                service(ScopeConverter::class),
+            ])
+        ->alias(AuthCodeRepositoryInterface::class, AuthCodeRepository::class)
+
+        // Security layer
+        ->set(OAuth2Provider::class)
+            ->args([
+                service(UserProviderInterface::class),
+                service(ResourceServer::class),
+                service(OAuth2TokenFactory::class),
+                null,
+            ])
+
+        ->set(OAuth2EntryPoint::class)
+
+        ->set(OAuth2Listener::class)
+            ->args([
+                service(TokenStorageInterface::class),
+                service(AuthenticationManagerInterface::class),
+                service('league.oauth2-server.psr_http_factory'),
+                service(OAuth2TokenFactory::class),
+                null,
+            ])
+
+        ->set(GrantConfigurator::class)
+            ->args([
+                tagged_iterator('league.oauth2-server.authorization_server.grant'),
+            ])
+
+        // League authorization server
+        ->set(AuthorizationServer::class)
+            ->args([
+                service(ClientRepositoryInterface::class),
+                service(AccessTokenRepositoryInterface::class),
+                service(ScopeRepositoryInterface::class),
+                null,
+                null,
+            ])
+            ->configurator(service(GrantConfigurator::class))
+        ->alias('league.oauth2-server.authorization_server', AuthorizationServer::class)
```

Comment:
Are these aliases useful?

Reply:
I think having good old named identifiers make sense for a bundle (distributed). 
But it should be the other way around: class-named services (aka autowiring types) should be aliases, not the opposite.
Let's keep this as is and revisit in another PR :)